src/common/guac_surface.c

```diff
@@ -437,8 +437,8 @@ static void __guac_common_surface_put(unsigned char* src_buffer, int src_stride,
 
     int x, y;
 
-    int min_x = rect->width - 1;
```

Comment:
The min_x and min_y values are set to valid values  if line 465 is true (old_color != new_color), which will be used in the width/heigh calculations.

If no pixels changed, the rect will be set to height/width of 0.

The odd-case which was not handled is if rect->width = rect->height = 1 and the pixel did not change. 
This caused the initial values to be evaluated as a true refresh at 
line 485 if (max_x >= min_x && max_y >= min_y)